packaging/nsis_settings.nsi

```diff
@@ -0,0 +1,47 @@
+!include "MUI2.nsh"
+SetCompressor lzma
+SetCompressorDictSize 128
+
+RequestExecutionLevel admin
+Unicode true
+!define PRODUCT "Mantid Imaging"
+
+Name "${PRODUCT}"
+OutFile "dist\${PRODUCT} Setup.exe"
+
+InstallDir "$PROGRAMFILES64\${PRODUCT}"
+
+!define MUI_ICON "../images/mantid_imaging_unstable_64px.ico"
+
+!insertmacro MUI_PAGE_LICENSE "../LICENSE.txt"
+!insertmacro MUI_PAGE_DIRECTORY
+!insertmacro MUI_PAGE_INSTFILES
+
+!insertmacro MUI_UNPAGE_CONFIRM
+!insertmacro MUI_UNPAGE_INSTFILES
+
+Section
+	SetShellVarContext all
+	SetOutPath $INSTDIR
+	File /r dist\MantidImaging\*
+	CreateDirectory "$SMPROGRAMS\${PRODUCT}"
+	CreateShortCut "$SMPROGRAMS\${PRODUCT}\${PRODUCT}.lnk" "$INSTDIR\MantidImaging.exe" \
+	   "" "$INSTDIR\MantidImaging.exe" 0
+
+	WriteUninstaller "$INSTDIR\uninstall.exe"
+
+	WriteRegStr SHCTX "Software\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT}" "DisplayName" "${PRODUCT}"
+	WriteRegStr SHCTX "Software\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT}" "UninstallString" '"$INSTDIR\uninstall.exe"'
+	WriteRegDWORD SHCTX "Software\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT}" "NoModify" 1
+	WriteRegDWORD SHCTX "Software\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT}" "NoRepair" 1
+SectionEnd
+
+Section "Uninstall"
+	SetShellVarContext all
+	Delete "$SMPROGRAMS\${PRODUCT}\${PRODUCT}.lnk"
+	Delete $INSTDIR\uninstall.exe
+	RMDir /r "$INSTDIR"
```

Comment:
`$INSTDIR` here is just the directory from which you run uninstall.exe, so it's not very safe to do this, e.g. someone may copy the uninstaller somewhere else before running it (this has actually happened to me before, when a user moved an uninstaller to their desktop before running it and it deleted their desktop folder).
See the warning here:
https://nsis.sourceforge.io/Reference/RMDir

Reply:
Filed as #1772